### PR TITLE
Add Parser

### PR DESCRIPTION
**Pull request by hwc0419**

### CODE REVIEW

File: src/main/java/seedu/duke/SyntaxAnalyser.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
+package seedu.duke;
+
+import java.util.ArrayList;
+import java.util.HashMap;
```
Comment (runxinghuan) [inline src/main/java/seedu/duke/SyntaxAnalyser.java:4]: These two imports are not used. I suggest deleting them.

**Review by runxinghuan — COMMENTED** [review]
There are a few code style errors. I suggest using ctrl+alt+l for auto indentation

File: src/main/java/seedu/duke/Parser.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package seedu.duke;
+
+public class Parser {
+    private String COMMAND_NAME;
+    private String[] argumentTokens = {};
+
+    public Parser(String userInput) throws IllegalCommandException, ArgumentMismatchException, BadTokenException {
+        //default case: User enters a valid command, the command expects no arguments
+        final String[] NAME_ARGSTR_PAIR = parseIntoNameAndArgumentString(userInput); //NAME_ARGSTR_PAIR is either size 1 or 2
```
Comment (runxinghuan) [inline src/main/java/seedu/duke/Parser.java:9]: This line is too long. Maybe move the comment to a new line?

File: src/main/java/seedu/duke/Formatter.java [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+     *
+     * @param lineNum the line number which does not follow the file format of cache file
+     */
+    public static void printFileCorruptionError(int lineNum) {
+        System.out.println("Cache file is corrupted at line " + lineNum);
+        System.out.println("Failed to load cache");
+        System.out.println("Program will start with empty list");
+    }
+
+//    public static void printMatches(String userQuery) {
```
Comment (HenryGan138) [inline src/main/java/seedu/duke/Formatter.java:236]: All the comment lines can be removed for clarity.

File: src/main/java/seedu/duke/Formatter.java [hunk 4]
```diff
@@ -0,0 +1,247 @@
+package seedu.duke;
+
+public class Formatter {
+    public static final String footballIcon = "⚽";
```
Comment (HenryGan138) [inline src/main/java/seedu/duke/Formatter.java:4]: This is really cute and nice design, I can add that icon into our code.

File: src/main/java/seedu/duke/Formatter.java [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+    }
+
+    /**
+     * Prints an error message stating the number of arguments given versus the number of arguments expected for given command
+     *
+     * @param correctArgumentCount the expected number of arguments for given command
+     * @param userArgumentCount    the given number of arguments read from user input
+     * @param userCommandName      the command name read from user input
+     */
+    public static void printErrorArgumentsMismatch(String userCommandName, int userArgumentCount, int correctArgumentCount) {
```
Comment (HenryGan138) [inline src/main/java/seedu/duke/Formatter.java:183]: The indentation seems incorrect

File: src/main/java/seedu/duke/SyntaxAnalyser.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    /**
+     * Returns true if argument type of all arguments are correct, false otherwise
+     *
+     * @param COMMAND_NAME   the command name of given command
+     * @param argumentTokens the list of arguments parsed from user input
+     * @return True if all argument types are correct
+     */
+    public static boolean validateTokens(String COMMAND_NAME, String[] argumentTokens) {
```
Comment (HenryGan138) [inline src/main/java/seedu/duke/SyntaxAnalyser.java:71]: You seem mispush some other function into the code.

File: src/main/java/seedu/duke/CommandList.java [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+//        System.out.println("Function");
+
+        String selectedDirection = userCommandReader.getArgumentTokens()[0];
+        int selectedDirectionIndex =  Integer.parseInt(selectedDirection);
+        boolean isScoreGoal = goalCheck(Ai.getAiDirection(), selectedDirectionIndex);
+
+        Formatter.printGoalAfterShot(isScoreGoal);
+    }
+
+//    /**
```
Comment (HenryGan138) [inline src/main/java/seedu/duke/CommandList.java:36]: Same to other functions, these hunk could be removed.

File: src/main/java/seedu/duke/SyntaxAnalyser.java (hunk 6 above)
Comment (hwc0419) [reply src/main/java/seedu/duke/SyntaxAnalyser.java:71]: can you elaborate on this

**Pull request merged**